rocAL/source/meta_data/label_reader_file_list.cpp

```diff
@@ -0,0 +1,128 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "label_reader_file_list.h"
+
+#include <string.h>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <utility>
+
+#include "commons.h"
+#include "exception.h"
+#include "filesystem.h"
+
+using namespace std;
+
+LabelReaderFileList::LabelReaderFileList() {
+    _src_dir = nullptr;
+    _entity = nullptr;
+    _sub_dir = nullptr;
+}
+
+void LabelReaderFileList::init(const MetaDataConfig& cfg, pMetaDataBatch meta_data_batch) {
+    _file_list_path = cfg.path();
+    _output = meta_data_batch;
+}
+
+bool LabelReaderFileList::exists(const std::string& sample_name) {
+    return _map_content.find(sample_name) != _map_content.end();
+}
+
+void LabelReaderFileList::add(std::string sample_name, int label) {
+    pMetaData info = std::make_shared<Label>(label);
+    if (exists(sample_name)) {
+        WRN("Entity with the same name exists")
+        return;
+    }
+    _map_content.insert(pair<std::string, std::shared_ptr<Label>>(sample_name, info));
+}
+
+void LabelReaderFileList::print_map_contents() {
+    std::cerr << "\nMap contents: \n";
+    for (auto& elem : _map_content) {
+        std::cerr << "Name :\t " << elem.first << "\t ID:  " << elem.second->get_labels()[0] << std::endl;
+    }
+}
+
+void LabelReaderFileList::release() {
+    _map_content.clear();
+}
+
+void LabelReaderFileList::release(std::string __sample) {
+    if (!exists(__sample)) {
+        WRN("ERROR: Given not present in the map" + __sample);
+        return;
+    }
+    _map_content.erase(__sample);
+}
+
+void LabelReaderFileList::lookup(const std::vector<std::string>& sample_names) {
+    if (sample_names.empty()) {
+        WRN("No image names passed")
+        return;
+    }
+    if (sample_names.size() != static_cast<unsigned>(_output->size()))
+        _output->resize(sample_names.size());
+    for (unsigned i = 0; i < sample_names.size(); i++) {
+        auto sample_name = sample_names[i];
+        auto it = _map_content.find(sample_name);
+        if (_map_content.end() == it)
+            THROW("Label Reader File List ERROR: Given name not present in the map " + sample_name)
+        _output->get_labels_batch()[i] = it->second->get_labels();
+    }
+}
+
+void LabelReaderFileList::read_all(const std::string& _path) {
+    std::string _folder_path = _path;
+    if ((_sub_dir = opendir(_folder_path.c_str())) == nullptr)
+        THROW("ERROR: Failed opening the directory at " + _folder_path);
+    std::vector<std::string> entry_name_list;
+    std::string _full_path = _folder_path;
+    while ((_entity = readdir(_sub_dir)) != nullptr) {
+        std::string entry_name(_entity->d_name);
+        if (strcmp(_entity->d_name, ".") == 0 || strcmp(_entity->d_name, "..") == 0) continue;
+        entry_name_list.push_back(entry_name);
+    }
+    std::sort(entry_name_list.begin(), entry_name_list.end());
```

Comment:
Lines 99 - 109 is not used, Please check

Reply:
Removed L99-109